crates/sui-types/src/event.rs

```diff
@@ -124,35 +127,83 @@ pub enum Event {
         package_id: ObjectID,
     },
     /// Transfer objects to new address / wrap in another object / coin
+    CoinBalanceChange {
+        package_id: ObjectID,
+        transaction_module: Identifier,
+        sender: SuiAddress,
+        change_type: BalanceChangeType,
+        owner: Owner,
+        coin_type: String,
+        coin_object_id: ObjectID,
+        version: SequenceNumber,
+        amount: i128,
+    },
+    /// Epoch change
+    EpochChange(EpochId),
+    /// New checkpoint
+    Checkpoint(CheckpointSequenceNumber),
+
+    /// Object level event
+    /// Transfer objects to new address / wrap in another object
     TransferObject {
         package_id: ObjectID,
         transaction_module: Identifier,
         sender: SuiAddress,
         recipient: Owner,
+        object_type: String,
+        object_id: ObjectID,
+        version: SequenceNumber,
+    },
+
+    /// Object level event
+    /// Object mutated.
+    MutateObject {
+        package_id: ObjectID,
+        transaction_module: Identifier,
+        sender: SuiAddress,
+        object_type: String,
         object_id: ObjectID,
         version: SequenceNumber,
-        type_: TransferType,
-        amount: Option<u64>,
     },
     /// Delete object
     DeleteObject {
         package_id: ObjectID,
         transaction_module: Identifier,
         sender: SuiAddress,
         object_id: ObjectID,
```

Comment:
object_type here as well?

Reply:
The object type is probably not relevant after the object is deleted?

Reply:
I was thinking of how ppl would use the events, for example ppl might want to listen to deletion events related to certain types, like a NFT or some type of tokens, to track objects of this kind without look-up each time, does that make sense?

Reply:
ok that make sense, I will add it

Reply:
Adding the object type turns out to be a big change, it require changes to `ObjectRuntimeState` to retrieve the type, we should do it in a subsequent PR....

created an issue to track this https://github.com/MystenLabs/sui/issues/5599